kong/db/schema/init.lua

```diff
@@ -1110,16 +1110,15 @@ validate_fields = function(self, input)
   for k, v in pairs(input) do
     local err
     local field = self.fields[tostring(k)]
-    local is_ttl = k == "ttl" and self.ttl
     if field and field.type == "self" then
       local pok
       pok, err, errors[k] = pcall(self.validate_field, self, input, v)
       if not pok then
         errors[k] = validation_errors.SCHEMA_CANNOT_VALIDATE
         kong.log.debug(errors[k], ": ", err)
       end
-    elseif is_ttl then
-      kong.log.debug("ignoring validation on ttl field")
+    elseif self.unvalidated_fields[k]() then
```

Comment:
If DP is relying on this new-add code to prevent schema validation failure, that requires DP to upgrade before CP, which is against our CP upgrade first policy.

I think it still needs a mechanism in CP to remove unknown fields before send to CP

Reply:
You're right in this case that it also has to be make at CP side. 
But I personally think there is a little space to improve the design of compatibility. For example, DP might be self-adapted, when encountering unknown field, it could be ignored, when some fields miss, it could use the original value or default value. Maybe my idea is too rough, but it's essential for the compatibility to be robust.